numpyro/contrib/module.py

```diff
@@ -433,3 +435,143 @@ def random_haiku_module(
         _update_params(params, new_params, prior)
     nn_new = partial(nn.func, new_params, *nn.args[1:], **nn.keywords)
     return nn_new
+
+
+def nnx_module(name, nn_module):
+    """
+    Declare a :mod:`~flax.nnx` style neural network inside a
+    model so that its parameters are registered for optimization via
+    :func:`~numpyro.primitives.param` statements.
+
+    Given a flax NNX ``nn_module``, to evaluate the module, we directly call it.
+    In a NumPyro model, the pattern will be::
+
+        # Eager initialization outside the model
+        module = nn_module(...)
+
+        # Inside the model
+        net = nnx_module("net", module)
+        y = net(x)
+
+    :param str name: name of the module to be registered.
+    :param flax.nnx.Module nn_module: a pre-initialized `flax nnx` Module instance.
+    :return: a callable that takes an array as an input and returns
+        the neural network transformed output array.
+    """
+    try:
+        from flax import nnx
+    except ImportError as e:
+        raise ImportError(
+            "Looking like you want to use flax.nnx to declare "
+            "nn modules. This is an experimental feature. "
+            "You need to install the latest version of `flax` to use this feature. "
+            "It can be installed with `pip install git+https://github.com/google/flax.git`."
+        ) from e
+
+    module_key = name + "$params"
+    module_params = numpyro.param(module_key)
+
+    if module_params is None:
+        graph_def, params_state, other_state = nnx.split(
+            nn_module, nnx.Param, nnx.filterlib.to_predicate(nnx.Not(nnx.Param))
+        )
+
+        params = nnx.to_pure_dict(params_state)
+
+        if params:
+            module_params = numpyro.param(module_key, params)
+
+        state_dict = {"state": params_state}
+
+        if any(state_dict.values()):
```

Comment:
Use similar logic as eager_params above, but for eager_other_state

Reply:
Do you mean like 

```python
    eager_other_params = nnx.to_pure_dict(eager_other_state)
    if eager_other_params:
        module_other_params = numpyro.param(name + "$params")
    if module_other_params is None:
        module_other_params = numpyro.param(name + "$params", eager_other_params)
``` 

?

Reply:
See my other comment on the role of mutable state during svi run.